### PR TITLE
Intialization of cdn media urls 

### PR DESCRIPTION
**Pull request by mahmed0715**
Fix https://github.com/Plant-for-the-Planet-org/treecounter-app/issues/1826
partially working, the urls are not ready yet.

### CODE REVIEW

**Comment by sagararyal** [comment]
@mahmed0715 works fine on web.

On Mobile the URLs do not resolve as app does not make api call to /config

Screen Shot 2020-01-20 at 10 14 02 PM

**Comment by mahmed0715** [comment]
@sagararyal The PR is updated with the correct formatted url from api, its good with test backend for now. Jorgo will deploy to prod server.
We can merge I think..

File: app/selectors/index.js [hunk 1]
```diff
@@ -46,6 +47,7 @@ export const pledgesSelector = state => getPledges(state);
 export const postedPledgesSelector = state => getPostedPledges(state);
 export const currenciesSelector = state => getCurrencies(state);
 export const getCurrency = state => getGlobalCurrency(state);
+export const getCdnMediaUrl = state => getCdnMedia(state);
```
Comment (norbertschuler) [inline app/selectors/index.js]: I am new to redux, therefore I would like to ask how is this definition related to the declaration of the function with the same name in `fetchLocation`?

File: app/components/App/TreeCounter.native.js [hunk 2]
```diff
@@ -11,10 +11,12 @@ import { bindActionCreators } from 'redux';
 import { SafeAreaView } from 'react-navigation';
 import { initLocale } from '../../actions/getLocale.native.js';
 import { fetchCurrencies } from '../../actions/currencies';
+import { fetchConfig } from '../../actions/fetchLocation';
 class App extends Component {
   componentDidMount() {
     initLocale();
     this.props.fetchCurrencies();
+    this.props.fetchConfig();
```
Comment (norbertschuler) [inline app/components/App/TreeCounter.native.js]: Why do you also load the config within `fetchLocation`? And don’t we need to load the config for the web client?

Comment (norbertschuler) [reply app/components/App/TreeCounter.native.js]: I just recognized `fetchLocation` is a function currently only loaded from the web client, so it implicitly load the config. But I would suggest not doing any API calls implicitly within the code of other API calls.

Comment (norbertschuler) [reply app/components/App/TreeCounter.native.js]: Changed it to make `fetchConfig` independent from `fetchLocation` for the web client.

File: app/selectors/index.js (hunk 1 above)
Comment (norbertschuler) [reply app/selectors/index.js]: If I compare this with currencies `getCurrency` is used from this file and there is not local function `getCurrency` in https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/develop/app/actions/currencies.js?

Comment (norbertschuler) [reply app/selectors/index.js]: I think you saved the cdnMedia config with redux, but never read the value again from redux. You just store it as a local variable within `app/actions/fetchLocation.js` and use it from there, right?

Comment (mahmed0715) [reply app/selectors/index.js]: yes. true. i was intended to use redux value but it was not needed for the need, but I think this might be needed next time when we want those values . but kept as is.

**Comment by sagararyal** [comment]
Looks good! test and staging do not work because there are no media in that bucket.
